baybe/searchspace/core.py

```diff
@@ -284,6 +288,32 @@ def n_tasks(self) -> int:
         except StopIteration:
             return 1
 
+    def get_comp_rep_parameter_indices(self, name: str, /) -> tuple[int, ...]:
+        """Find a parameter's column indices in the computational representation.
+
+        Args:
+            name: The name of the parameter whose columns indices are to be retrieved.
+
+        Raises:
+            ValueError: If no parameter with the provided name exists.
+
+        Returns:
+            A tuple containing the integer indices of the columns in the computational
+            representation associated with the parameter. When the parameter is not part
+            of the computational representation, an empty tuple is returned.
+        """
+        if name not in (p.name for p in self.parameters):
+            raise ValueError(
+                f"There exists no parameter named '{name}' in the search space."
+            )
+
+        # TODO: The "startswith" approach is not ideal since it relies on the implicit
```

Comment:
This is in fact a subtle but **critical** bug I noticed. Fails when the parameter names have overlapping substrings themselves! Consider:
```python
from baybe.parameters.numerical import NumericalDiscreteParameter
from baybe.searchspace.core import SearchSpace

searchspace = SearchSpace.from_product(
    [
        NumericalDiscreteParameter("param", [0, 1]),
        NumericalDiscreteParameter("parameter", [0, 1]),
    ]
)
print(searchspace.get_comp_rep_parameter_indices("param"))
```
This will incorrectly give you `(0, 1)` instead of `(0,)`! So substring matching is inherently flawed!

Alternatives I see:
* Instead of substring matching, we could check full strings, but that requires knowing the comp_rep parameter column names. And that would result in a bad architecture, since it would couple the logic here to the naming conventions when creating the comp_rep (e.g. one-hot encoded columns etc). Not great
* We could just monitor where things end up when we build the comp_df and store that information. Drawback: one additional private attribute + needs a little piece of code implementing that logic. But other than that, clean solution.
* Do you have better ideas?

To your question why it is written here: because this is the place where the code breaks.

Reply:
id go for exact matche sand not startswith, for dicrete ones via `p.comp_df.columns` and for conti ones by name

Reply:
Have opened the corresponding issue #352  for a mini bugfix PR, so that I can close here